guide/01-getting-started/package-json.md

```diff
@@ -120,6 +120,26 @@ When `exports` is provided, usually there's no need to specify the `main` field
 
 To learn more about the `exports` field, see [Node.js documentation on package exports](https://nodejs.org/api/packages.html#package-entry-points).
 
+## The `files` field
```

Comment:
what about npmignore?

imo npmignore is far superior to the files field, because if you forget to put something in npmignore, nothing's broken - but if you forget to put something in files, all of your downstream consumers are broken.

Reply:
It might be a bit too package manager specific while this guide is trying to be more package manager agnostic (while using npm mostly to provide quick verification steps or "this is where you'd find the common denominator"). Though we can add an appendix/separate chapter about this package-manager specific things and pros/cons.

> because if you forget to put something in npmignore, nothing's broken 

It kind of depends on what is considered broken, e.g. accidentally publishing credentials stored in a file may cause more irreversible damage than accidentally breaking users.

Reply:
I think what we are looking for is not to endorse any pattern, neither `files` nor `.npmignore`. Rather it'd better to use wording along the lines "to do this, this is what people usually does" (in the main example) + a pointer to "there are also some maintainers who prefer this". The reader can decide if they want to follow the mass or read through the pros/cons of the less common option and make an informed choice.

Reply:
In general I think that's fine, but the reason files is more common is because fewer people are aware of the alternative, or of the cons of files. I think node has an obligation to not just document reality, but encourage improvements.

Reply:
Added a TODO to the multi-file package guide to document .npmignore (this page seems less right to put that in because it's specific about pacakge.json).